File: Trinity.Runtime/platform_src/ios/elastOS/Classes/ContactNotifier/comm/AcceptFriendCommand.swift
public class AcceptFriendCommand : CarrierCommand {
    private let helper: CarrierHelper
    private let contactCarrierUserID: String
    private let completionListener: CarrierHelper.onCommandExecuted

    init(helper: CarrierHelper, contactCarrierUserID: String, completionListener: @escaping CarrierHelper.onCommandExecuted) {
        self.helper = helper
        self.contactCarrierUserID = contactCarrierUserID
        self.completionListener = completionListener
    }

    public func executeCommand() {
        Log.i(ContactNotifier.LOG_TAG, "Executing accept friend command")
        do {
            try helper.carrierInstance!.acceptFriend(with: contactCarrierUserID)

            completionListener(true, nil)
        }
        catch (let error) {
            print(error)
            completionListener(false, error.localizedDescription)
        }
    }
}

File: Trinity.Runtime/platform_src/android/app/src/main/java/org/elastos/trinity/runtime/contactnotifier/comm/AcceptFriendCommand.java
package org.elastos.trinity.runtime.contactnotifier.comm;

import android.util.Log;

import org.elastos.trinity.runtime.contactnotifier.ContactNotifier;
import org.json.JSONObject;

public class AcceptFriendCommand implements CarrierCommand {
    private CarrierHelper helper;
    private String contactCarrierUserID;
    private CarrierHelper.OnCommandExecuted completionListener;

    AcceptFriendCommand(CarrierHelper helper, String contactCarrierUserID, CarrierHelper.OnCommandExecuted completionListener) {
        this.helper = helper;
        this.contactCarrierUserID = contactCarrierUserID;
        this.completionListener = completionListener;
    }

    @Override
    public void executeCommand() {
        Log.i(ContactNotifier.LOG_TAG, "Executing accept friend command");
        try {
            helper.carrierInstance.acceptFriend(contactCarrierUserID);

            completionListener.onCommandExecuted(true, null);
        }
        catch (Exception e) {
            e.printStackTrace();
            completionListener.onCommandExecuted(false, e.getLocalizedMessage());
        }
    }
}